plugins/rapid7_insightvm/unit_test/test_tag_assets.py

```diff
@@ -0,0 +1,88 @@
+import sys
+import os
+
+sys.path.append(os.path.abspath("../"))
+
+from unittest import TestCase
+from unittest.mock import patch
+from unit_test.util import Util
+from komand_rapid7_insightvm.actions.tag_assets import TagAssets
+from insightconnect_plugin_runtime.exceptions import PluginException
+
+
+@patch("requests.sessions.Session.put", side_effect=Util.mocked_requests)
+class TestTagAssets(TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.action = Util.default_connector(TagAssets())
+
+    def setUp(self) -> None:
+        self.params = {
+            "tag_id": 1,
+            "asset_ids": [1, 2, 3],
+            "tag_source": "CUSTOM",
+            "tag_type": "Custom",
+            "tag_name": "Example Tag",
+        }
+
+    def test_tag_asset(self, mock_put):
+        expected = {"success": True}
+        actual = self.action.run(self.params)
+        self.assertEqual(actual, expected)
+
+    def test_tag_asset_bad_id(self, mock_put):
+        self.params["tag_id"] = 2
+        cause = "Malformed JSON received along with a status code of Not Found"
+        assistance = (
+            "Verify your connection is pointing to your local console and not "
+            "`exposure-analytics.insight.rapid7.com` Ensure that the requested resource exists."
+        )
+        data = "The ID 2 is not available. Enter a different ID for this tag."
+        with self.assertRaises(PluginException) as e:
+            self.action.run(self.params)
+        self.assertEqual(cause, e.exception.cause)
+        self.assertEqual(assistance, e.exception.assistance)
+        self.assertEqual(data, e.exception.data)
```

Comment:
Well personally I don't like single letter variables, but it's up to you 😃 
```suggestion
        with self.assertRaises(PluginException) as error:
            self.action.run(self.params)
        self.assertEqual(cause, error.exception.cause)
        self.assertEqual(assistance, error.exception.assistance)
        self.assertEqual(data, error.exception.data)
```